test/redis_mutex_without_mock_test.exs

```diff
@@ -1,19 +1,53 @@
 defmodule RedisMutexWithoutMockTest do
-  use ExUnit.Case
-  use RedisMutex, lock_module: RedisMutex.Lock
-  doctest RedisMutex
+  use ExUnit.Case, async: false
 
   @moduletag :redis_dependent
 
-  describe "with_lock" do
+  defmodule RedisMutexUser do
+    use RedisMutex, lock_module: RedisMutex.Lock
+
+    def two_threads_lock do
+      with_lock("two_threads_lock") do
+        start_time = DateTime.utc_now()
+        end_time = DateTime.utc_now()
+        {start_time, end_time}
+      end
+    end
+
+    def two_threads_one_loses_lock do
+      with_lock("two_threads_one_loses_lock", 500) do
+        start_time = DateTime.utc_now()
+        :timer.sleep(1000)
+        end_time = DateTime.utc_now()
+        {start_time, end_time}
+      end
+    rescue
+      RedisMutex.Error -> :timed_out
+    end
+
+    def long_running_task do
+      with_lock("two_threads_lock_expires", 10_000, 250) do
+        :timer.sleep(10_000)
+      end
+    end
+
+    def quick_task do
+      with_lock("two_threads_lock_expires", 1000, 500) do
+        "I RAN!!!"
+      end
+    end
+  end
+
+  describe "with_lock/4" do
+    setup do
+      start_supervised({RedisMutex, lock_module: RedisMutex.Lock})
+      :ok
+    end
+
     test "works with two tasks contending for the same lock, making one run after the other" do
       res =
         run_in_parallel(2, 5000, fn ->
```

Comment:
is this 5 seconds? cause that seems a bit of a slow test and I am sure 500ms is plenty?